### PR TITLE
[Testing] Add load-testing foundation

### PR DESCRIPTION
**Pull request by okdas**
### Human Summary

Adds load testing functionality for Poktroll stack.

### Summary generated by Reviewpad on 19 Jan 24 00:40 UTC
This pull request includes the following changes:

1. The file "load_test_1.md" has been renamed to "load_testing_plan.md". The title of the file has also been changed from "Load Test 1 - Plan" to "Load Testing - Plan".

2. The file "sendJsonRPCRequest.js" has been modified to add functionality for sending JSON-RPC requests to Anvil.

3. The file "load-testing/config/index.js" has been added. It exports environment configuration variables and a function for generating HTML and text summaries of test data. It also includes options for load testing and commented sections for executing the test script in Grafana Cloud and enabling the use of the Browser API.

4. The file "load-testing/scenarios/requestBlockNumberEtherium.js" has been added. It exports a function that sends an Ethereum JSON-RPC request using the "sendJsonRPCRequest" function and receives a response. It also includes commented-out sections for adding tags to the request and additional checks.

5. The file "load-testing/README.md" has been added.

6. The Makefile has been modified to include a new target called "load_test_simple" that runs a load test using the "k6" tool. The load test script can be found at "load-testing/tests/appGateServerEtherium.js".

7. The ".gitignore" file has been modified to add "summary.html" to the gitignore.

8. The "openapi.yml" file has been modified to define object types and remove lines related to `poktroll.service.MsgAddServiceResponse`.

9. The file "load-testing/README.md" has been added to introduce load testing for the Poktroll infrastructure, explaining the load testing process, dependencies, how to run tests using the k6 tool, and understanding load test output.

10. The file "load-testing/tests/appGateServerEtherium.js" has been added to export variables and functions, import modules, and define a default function for executing a scenario and adding think time.

11. The file "load-testing/scenarios/anvil.js" has been added to export a function for producing HTML reports, import modules and functions, export an object with options, and define a default function for executing a scenario with a delay.

Please let me know if you need more information or further assistance.

## Issue

- #29 
- #30 

## Type of change

Select one or more:

- [x] New feature, functionality or library
- [ ] Bug fix
- [ ] Code health or cleanup
- [x] Documentation
- [ ] Other (specify)

## Testing

- [ ] **Run all unit tests**: `make go_develop_and_test`
- [ ] **Run E2E tests locally**: `make test_e2e`
- [ ] **Run E2E tests on DevNet**: Add the `devnet-test-e2e` label to the PR. This is VERY expensive, only do it after all the reviews are complete.

## Sanity Checklist

- [x] I have tested my changes using the available tooling
- [x] I have performed a self-review of my own code
- [x] I have commented my code, updated documentation and left TODOs throughout the codebase

### CODE REVIEW

File: summary.html
Comment (okdas) [inline summary.html:1]: Evidently, I accidentally committed this to main..

File: load-testing/tests/appGateServerEtherium.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+// Our custom handleSummary produces HTML report with CLI output and some default configuration.
```
Comment (Olshansk) [inline load-testing/tests/appGateServerEtherium.js:1]: `app_gate_server.js`?

That way we can add logic in the same place.

File: load-testing/README.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+### HTML Report
+
+An HTML report is generated in the execution directory. Open it in your default browser:
+
+```
+open summary.html
+```
+
+### Sophisticated Reporting
```
Comment (Olshansk) [inline load-testing/README.md:86]:
```suggestion

### Adding a new load test

TODO(@okdas): Add link to PR next time a new type of load test is added.

### Sophisticated Reporting
```

**Review by Olshansk — CHANGES_REQUESTED** [review]
This was super easy to review and looks like exactly what we need. Excited to keep this going!

File: load-testing/tests/appGateServerEtherium.js (hunk 1 above)
Comment (okdas) [reply load-testing/tests/appGateServerEtherium.js:1]: There will be different tests for app gate server, not just one.

File: load-testing/modules/etheriumRequests.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
+// Import necessary modules
+import http from 'k6/http';
+import { check } from 'k6';
+
+// Function to send a generic Ethereum JSON-RPC request to Anvil
+export function sendEthereumRequest(baseUrl, method, params = [], tags = {}) {
```
[the remark this answers is not in the document]
Comment (okdas) [reply load-testing/modules/etheriumRequests.js]: good call!

File: load-testing/modules/etheriumRequests.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    let passed = check(response, {
+        "is status 200": (r) => r.status === 200,
+        "is successful JSON-RPC response": (r) => {
+            let jsonResponse = JSON.parse(r.body);
+            // Check for 'result' and 'id', and ensure 'error' is not present
+            return jsonResponse.hasOwnProperty("result") && jsonResponse.hasOwnProperty("id") && !jsonResponse.hasOwnProperty("error");
+        }
+    }, tags);
+
+    // TODO: remove this or make configurable. But for now, let's output all debug info if the request doesn't pass checks.
```
[the remark this answers is not in the document]
Comment (okdas) [reply load-testing/modules/etheriumRequests.js]: Payload is included in error output as a part of `response`.

File: load-testing/README.md [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+- [Understanding Output](#understanding-output)
+  - [CLI Output](#cli-output)
+  - [HTML Report](#html-report)
+  - [Sophisticated Reporting](#sophisticated-reporting)
+- [Troubleshooting](#troubleshooting)
+- [File structure](#file-structure)
+
+## Overview
+
+We use [k6](https://k6.io/) for load testing. For detailed information about k6 internals and best practices, please refer to the [k6 documentation](https://grafana.com/docs/k6/latest/).
```
[the remark this answers is not in the document]
Comment (okdas) [reply load-testing/README.md]: I was thinking doesn't apply to Markdown. :)

**Pull request merged**